src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int MAX_START_LENGTH = 10;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[MAX_START_LENGTH];
+    }
+
+    private void grow() {
+        T[] newArray = (T[]) new Object[elements.length + elements.length / 2];
+        System.arraycopy(elements, 0, newArray, 0, size);
+        elements = newArray;
+    }
+
+    private void exception(int index) {
```

Comment:
```suggestion
    private void checkIndex(int index) {
```

What about method naming? https://mate-academy.github.io/style-guides/java/java.html#s5.2.3-method-names